src/components/TodoFilter/TodoFilter.tsx

```diff
@@ -1,30 +1,64 @@
-export const TodoFilter = () => (
-  <form className="field has-addons">
-    <p className="control">
-      <span className="select">
-        <select data-cy="statusSelect">
-          <option value="all">All</option>
-          <option value="active">Active</option>
-          <option value="completed">Completed</option>
-        </select>
-      </span>
-    </p>
+import React from 'react';
+import { Category } from '../../types/enums/Category';
 
-    <p className="control is-expanded has-icons-left has-icons-right">
-      <input
-        data-cy="searchInput"
-        type="text"
-        className="input"
-        placeholder="Search..."
-      />
-      <span className="icon is-left">
-        <i className="fas fa-magnifying-glass" />
-      </span>
+interface Props {
+  query: string;
+  setQuery: (query: string) => void;
+  category: string;
+  setCategory: (category: string) => void;
+}
+export const TodoFilter: React.FC<Props> = props => {
+  const { query, setQuery, category, setCategory } = props;
 
-      <span className="icon is-right" style={{ pointerEvents: 'all' }}>
-        {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
-        <button data-cy="clearSearchButton" type="button" className="delete" />
-      </span>
-    </p>
-  </form>
-);
+  let isResetVisible = category !== Category.All || query !== '';
```

Comment:
State Management Issue: The `isResetVisible` variable is being modified directly, which won't trigger a re-render. Consider using a state hook (e.g., `useState`) to manage `isResetVisible` so that the component updates correctly when its value changes.